kong_test.go

```diff
@@ -1542,3 +1542,106 @@ func TestPassthroughCmdOnlyStringArgs(t *testing.T) {
 	_, err := kong.New(&cli)
 	require.EqualError(t, err, "<anonymous struct>.Command: passthrough command command [<args> ...] must contain exactly one positional argument of []string type")
 }
+
+func TestStringPointer(t *testing.T) {
+	var cli struct {
+		Foo *string
+	}
+	k, err := kong.New(&cli)
+	require.NoError(t, err)
+	require.NotNil(t, k)
+	ctx, err := k.Parse([]string{"--foo", "wtf"})
+	require.NoError(t, err)
+	require.NotNil(t, ctx)
+	require.NotNil(t, cli.Foo)
+	require.Equal(t, "wtf", *cli.Foo)
+}
+
+func TestStringPointerNoValue(t *testing.T) {
```

Comment:
how about test for enum? We should be able to omit the flag even if it has an enum attribute, correct?

Reply:
Good catch. There were some bugs with enums. I added a few test cases and updated appropriately. Can you take another look.